pkg/controller/nodelifecycle/node_lifecycle_controller_test.go

```diff
@@ -2488,6 +2490,43 @@ func TestMonitorNodeHealthMarkPodsNotReadyRetry(t *testing.T) {
 			},
 			expectedPodStatusUpdates: 2, // One failed and one retry.
 		},
+		// Node created long time ago, with status updated by kubelet exceeds grace period.
+		// First monitorNodeHealth check will fail to list pods.
+		// Second monitorNodeHealth check will update pod status to NotReady (retry).
+		{
+			desc: "unsuccessful pod list, retry required",
+			fakeNodeHandler: &testutil.FakeNodeHandler{
+				Clientset: fake.NewSimpleClientset(&v1.PodList{Items: []v1.Pod{*testutil.NewPod("pod0", "node0")}}),
+			},
+			fakeGetPodsAssignedToNode: func() func(c *fake.Clientset) func(string) ([]*v1.Pod, error) {
```

Comment:
I don't get it - why you need func()?
Why it can't be:
```
func(c *fake.Clientset) func(string) ([]*v1.Pod, error) {
  // what you currently have?
}
```

Reply:
Good catch. I removed unneeded closure.